tensorflow/python/autograph/converters/control_flow.py

```diff
@@ -309,9 +309,8 @@ def visit_For(self, node):
     node = self.generic_visit(node)
     body_scope = anno.getanno(node, annos.NodeAnno.BODY_SCOPE)
     iter_scope = anno.getanno(node, annos.NodeAnno.ITERATE_SCOPE)
-
     loop_vars, undefined, _ = self._get_block_vars(
-        node, body_scope.bound | iter_scope.bound)
+        node, body_scope.bound | iter_scope.bound, iter_scope.bound)
 
     undefined_assigns = self._create_undefined_assigns(undefined)
 
```

Comment:
Once we do this, we also need to remove the iterate arg from e.g. here: `def body_name(iterate_arg_name):`. In essence, we're changing how the iterate variable is passed to the loop body.

Reply:
Yes I've tried to do this but If I remember correctly there was a side effect so I've not commited that part.
I need to check It again.

Reply:
Oh and what you want to add to `loop_vars` is not iter_scope.bound, but the `iter_arg_name` being created further below. So basically if you have `for x, y in something:` then you still want just: (1) one `itr` and a `x, y = itr` in the loop body.

So it's definitely a more invasive change.

Reply:
> remove the iterate arg from e.g. here: def body_name(iterate_arg_name)

This requires to modify the template right?